src/rust/engine/process_execution/src/nailgun/tests.rs

```diff
@@ -0,0 +1,211 @@
+  let req = super::construct_nailgun_server_request(
+    &NAILGUN_MAIN_CLASS.to_string(),
+    Vec::new(),
+    PathBuf::from(""),
+    Platform::None,
+  );
+  assert_eq!(req.argv[0], NAILGUN_MAIN_CLASS);
+  assert_eq!(req.argv[1..], ARGS_TO_START_NAILGUN);
+}
+
+#[test]
+fn creating_nailgun_client_request_inserts_port_as_an_env_var() {
+  let original_req = mock_nailgunnable_request(None);
+  let req = super::construct_nailgun_client_request(
+    original_req,
+    "".to_string(),
+    "".to_string(),
+    vec![],
+    1234,
+  );
+  assert_eq!(
+    req.env.get(NAILGUN_PORT_ENV_VAR_FOR_CLIENT),
+    Some(&String::from("1234"))
+  );
+}
+
+#[test]
+fn creating_nailgun_client_request_removes_jdk_home() {
+  let original_req = mock_nailgunnable_request(Some(PathBuf::from("some/path")));
+  let req = super::construct_nailgun_client_request(
+    original_req,
+    "".to_string(),
+    "".to_string(),
+    vec![],
+    1234,
+  );
+  assert_eq!(req.jdk_home, None);
+}
+
+#[test]
+fn nailgun_name_is_the_main_class() {
+  let main_class = "my.main.class".to_string();
+  let name = super::CommandRunner::calculate_nailgun_name(&main_class);
+  assert_eq!(name, format!("nailgun_server_{}", main_class));
+}
+
+fn materialize_with_jdk(
+  runner: &CommandRunner,
+  dir: PathBuf,
+  jdk_path: PathBuf,
+) -> Result<(), String> {
+  let executor = task_executor::Executor::new();
+  let materializer =
+    runner.materialize_workdir_for_server(dir, Some(jdk_path), EMPTY_DIGEST, WorkUnitStore::new());
+  executor.block_on(materializer)
+}
+
+#[test]
+fn materializing_workdir_for_server_creates_a_link_for_the_jdk() {
+  let workdir_base_tempdir = unique_temp_dir(std::env::temp_dir(), None);
+  let workdir_base = workdir_base_tempdir.path().to_owned();
+  let mock_jdk_dir = unique_temp_dir(std::env::temp_dir(), None);
+  let mock_jdk_path = mock_jdk_dir.path().to_owned();
+  let runner = mock_nailgun_runner(Some(workdir_base));
+  let nailgun_name = "mock_server".to_string();
+
+  let workdir_for_server = runner
+    .get_nailgun_workdir(&nailgun_name)
+    .expect("Error creating workdir for nailgun server");
+  println!("Workdir for server {:?}", &workdir_for_server);
+
+  // Assert that the materialization was successful
+  let materialization_result =
+    materialize_with_jdk(&runner, workdir_for_server.clone(), mock_jdk_path.clone());
+  assert_eq!(materialization_result, Ok(()));
+
+  // Assert that the symlink points to the requested jdk
+  let materialized_jdk_path = workdir_for_server.join(".jdk");
+  let materialized_jdk = read_link(materialized_jdk_path);
+  assert!(materialized_jdk.is_ok());
+  assert_eq!(materialized_jdk.unwrap(), mock_jdk_path);
+}
+
+#[test]
+fn materializing_workdir_for_server_replaces_jdk_link_if_a_different_one_is_requested() {
+  let workdir_base_tempdir = unique_temp_dir(std::env::temp_dir(), None);
+  let workdir_base = workdir_base_tempdir.path().to_owned();
+
+  let _ = workdir_base_tempdir.into_path();
```

Comment:
This means the dir won't be deleted at the end of the test - why not `.path()`?

Reply:
This was a leftover from debugging the test. Good catch, and I'm sorry it was left there.